realm/realm-library/src/main/cpp/CMakeLists.txt

```diff
@@ -118,9 +118,9 @@ endif()
 #     'best.m_nanoseconds' was declared here
 set(WARNING_CXX_FLAGS "-Wall -Wextra -pedantic -Wno-long-long -Wno-variadic-macros \
 -Wno-missing-field-initializers -Wmissing-declarations -Wno-error=uninitialized -Wno-error=maybe-uninitialized")
-set(REALM_COMMON_CXX_FLAGS "-DREALM_ANDROID -DREALM_HAVE_CONFIG -DPIC -pthread -fvisibility=hidden -std=c++14 -fsigned-char")
+set(REALM_COMMON_CXX_FLAGS "-DREALM_ANDROID -DREALM_HAVE_CONFIG -DREALM_HAVE_EPOLL -DPIC -pthread -fvisibility=hidden -std=c++14 -fsigned-char")
 if (build_SYNC)
-    set(REALM_COMMON_CXX_FLAGS "${REALM_COMMON_CXX_FLAGS} -DREALM_SYNC")
+    set(REALM_COMMON_CXX_FLAGS "${REALM_COMMON_CXX_FLAGS} -DREALM_SYNC -DREALM_ENABLE_SYNC")
```

Comment:
Remove the ` -DREALM_SYNC` and use `REALM_ENABLE_SYNC` in the `io_realm_internal_SharedRealm.cpp` as well. OS refactored the macro from `REALM_SYNC` to `REALM_ENABLE_SYNC` and we should do the same.